app/src/main/java/com/garethevans/church/opensongtablet/ProcessSong.java

```diff
@@ -951,14 +951,19 @@ String determineLineTypes(String string, Context c) {
         ArrayList<String> chordpositions = new ArrayList<>();
 
         // Set the start of the line as the first bit
-        chordpositions.add("0");
+        if (string.startsWith(".")) {
+            chordpositions.add("1");
+        } else {
+            chordpositions.add("0");
+        }
 
+        // IV - Comments do not seem to reflect the code logic?
         // In order to identify chords at the end of the line
         // (My method looks for a following space)
         // Add a space to the search string.
-        string += " ";
+        string = " " + string.substring(1) + " ";
```

Comment:
Found a bug here.  I'll amend the code directly.
If an empty string is passed (a blank line with no space), string.substring(1) is out of bounds.  Because it is called inside a try/catch method, it fails quietly, but no song is displayed.
I added a quick check for empty lines and made them default to a single space.

Reply:
Thanks.